### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by limtingzhi**
**Changes:**
- Update dependencies
- Add date-fns as peer dependencies
- Fix vulnerabilities
- Update target browsers list
- Update GitHub workflows
- Add pull request template

### CODE REVIEW

File: release.config.js [hunk 1]
```diff
@@ -35,6 +36,12 @@ module.exports = {
         ],
       },
     ],
+    [
+      '@semantic-release/exec',
+      {
+        publishCmd: 'git checkout -- package.json',
+      },
+    ],
```
Comment (moroine) [inline release.config.js:44]: This part, to not update the package.json version is required only when using mono-repo. 

If we don't want to update the version, maybe put `0.0.0` in `package.json`

Comment (limtingzhi) [reply release.config.js:44]: I think I will remove `@semantic-release/exec` instead since this is not a mono repo.

**Review by moroine — APPROVED** [review]

**Pull request merged**